github-rag/app.py

```
@@ -13,51 +15,117 @@
 def reset_chat():
     """Reset chat session and clean up resources"""
     try:
         st.session_state.messages = []
         st.session_state.context = None
-        gc.collect()
+        cleanup_session()
         logger.info("Chat session reset successfully")
     except Exception as e:
         logger.error(f"Error resetting chat: {str(e)}")
-        raise GitHubRAGError("Failed to reset chat session")
+        raise SessionError("Failed to reset chat session")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)